src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.cs

```diff
@@ -265,5 +261,31 @@ public void EndInit_DoesNotEnableEventRaisedEvents()
                 Thread.Sleep(WaitForExpectedEventTimeout);
             }
         }
+
+        [Fact]
+        public void DroppedWatcher_Collectible()
+        {
+            WeakReference watcher = CreateEnabledWatcher(TestDirectory);
+            File.Create(GetTestFilePath()).Dispose();
+            Assert.True(SpinWait.SpinUntil(() =>
+            {
+                GC.Collect();
+                return !watcher.IsAlive;
+            }, LongWaitTimeout));
+
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
why no inlining particularly?

Reply:
Because if it got inlined, a temporary local created by the JIT could end up rooting the instance on the stack indefinitely.

Reply:
Oh of course